### PR TITLE
fix test.sh (now it works)

### PR DESCRIPTION
**Pull request by vorburger**

### CODE REVIEW

**Comment by vorburger** [comment]
@rhuss merge this? :smile:

**Comment by vorburger** [comment]
This will conflict with #135, but this probably should go in before that.

File: test.sh [hunk 1]
```diff
@@ -5,10 +5,25 @@ cd java ; fish-pepper ; cd ..
...
+# Now rebuild incrementally, it should not re-download .m2
+s2i build --copy java/example fabric8/s2i-java fabric8/s2i-java-example --incremental
 
-# TODO how to do simplest possible HTTP GET http://localhost:8080 and fail/pass?
-# TODO how to best stop docker again
-# kill $!
+CONTAINER_ID=$(docker run -d -p 8080:8080 fabric8/s2i-java-example)
+
+# sleep 1s is required because after docker run returns, the container is up but our server may not quite be yet
+sleep 1
```
Comment (rhuss) [inline test.sh]: Is 1s good enough ?

File: test.sh [hunk 2]
```diff
@@ -5,10 +5,25 @@ cd java ; fish-pepper ; cd ..
...
+
+HTTP_REPLY=$(curl http://localhost:8080)
```
Comment (rhuss) [inline test.sh]: `-s` might be helpful here

File: test.sh [hunk 3]
```diff
@@ -5,10 +5,25 @@ cd java ; fish-pepper ; cd ..
...
+s2i build --copy java/example fabric8/s2i-java fabric8/s2i-java-example
 
-docker run -p 8080:8080 fabric8/s2i-java-example
+# Now rebuild incrementally, it should not re-download .m2
+s2i build --copy java/example fabric8/s2i-java fabric8/s2i-java-example --incremental
 
-# TODO how to do simplest possible HTTP GET http://localhost:8080 and fail/pass?
-# TODO how to best stop docker again
-# kill $!
+CONTAINER_ID=$(docker run -d -p 8080:8080 fabric8/s2i-java-example)
```
Comment (rhuss) [inline test.sh]: Not really sure in which context this test is run, but are you sure that 8080 on localhost is always free ? wouldn't it make sense to use a dynamic portmapping ?

Comment (rhuss) [reply test.sh]: Also what is if the docker daemon is running remotely ? (again not sure in which context this script is supposed to be running)

File: test.sh (hunk 1 above)
Comment (vorburger) [reply test.sh]: It "works for me", but let me make it 5s instead of 1s just to be on the safe side...

File: test.sh (hunk 2 above)
Comment (vorburger) [reply test.sh]: OK, but let me add "--silent --show-error", as "man curl" seems to recommend.

**Comment by rhuss** [comment]
Thanks ! I'm quite busy will add this asap ...

**Comment by vorburger** [comment]
@rhuss ping - any chance you / anyone could see if this is good and could go in now? 

I'm waiting for this one to go in my other pending PRs I would like to wrap up, once this one moves.

**Pull request merged**